### PR TITLE
[Website] Update documentations for 0.16.0

### PR DESCRIPTION
**Pull request by kszucs**
Used https://github.com/apache/arrow/pull/6407 to generate this. Doxygen is moved to docs/cpp-doxygen

### CODE REVIEW

**Comment by kou** [comment]
It seems that this doesn't include any documentation update.

**Comment by kszucs** [comment]
@kou I'm not sure what do you mean. The diff is huge, which makes it hard to follow.

**Comment by wesm** [comment]
@kszucs I am not sure we should delete the Doxygen? What is in Sphinx is not comprehensive

**Comment by kszucs** [comment]
I [wasn't sure](https://github.com/apache/arrow/pull/6407/files#diff-a35ecb896ea5aa4ce78422af5b5bcf07R34) either. Based on the [current documentation page](http://arrow.apache.org/docs/cpp/) doxygen is not reachable.

I can add the oxygen files as well. Shall we expose the oxygen docs in a different directory and a different link?

**Comment by wesm** [comment]
I think we should link to the Doxygen from the Sphinx project (you can put external links in the TOC tree). We can do that in a follow up change to the docs

**Comment by wesm** [comment]
https://issues.apache.org/jira/browse/ARROW-7835

**Comment by wesm** [comment]
There are broken links in the release blog post. Does anything else need to be done in this aside from restoring the Doxygen docs?

**Comment by wesm** [comment]
I restored the Doxygen to its own directory
